backend/main.py

```diff
@@ -1,34 +1,56 @@
-from fastapi import FastAPI
+from fastapi import FastAPI, HTTPException,Depends
 from db import get_database
 import json
 from model import *
+from typing import List
 import re
+from fastapi.middleware.cors import CORSMiddleware
 
 app = FastAPI()
 db = get_database()
 collection = db["apps"]
 
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=[""],
+    allow_credentials=True,
+    allow_methods=[""],
+    allow_headers=["*"],
+)
 
-@app.post("/add_item", response_model=str)
-async def add_item(item: AddItem):
-    # Insert the new item into the database
-    await collection.insert_one(item.model_dump())
+
+@app.post("/addItem", response_model=str)
+async def addItem(item: AddItem):
+    # Check if an item with the same name or ID already exists
+    existing_item = await collection.find_one({"$or": [{"name": item.name}, {"id": item.id}]})
+    
+    if existing_item:
+        raise HTTPException(status_code=400, detail="Item with the same name or ID already exists")
+
+    await collection.insert_one({"name": item.name, "icon": item.icon, "id": item.id})
     return "Item added successfully!"
 
 
-@app.get("/get_all_items")
-async def get_all_items():
-    items = await collection.find({}, {"_id": 0}).to_list()
+@app.get("/getAllItems")
+async def getAllItems(name: Optional[str] = None,id: Optional[str] = None):
+    query = {}
+
+    if name:
+        query["name"] = re.compile(name, re.IGNORECASE)
+    elif id:
+        query["id"] = re.compile(id, re.IGNORECASE)
+
+    items = await collection.find(query, {"_id": 0,"icon":0}).to_list()
     return items
 
 
-@app.get("/items/{appquery}", response_model=str)
-async def read_item(items: str):
-    winget_script = "winget install "
-    for i in items.split(","):
+@app.get("/items/{appQuery}", response_model=str)
+async def readItem(appQuery: str):
+    wingetScript = "winget install "
+    for itemName in appQuery.split(","):
         # Removing spaces from the string
-        i = i.strip().replace(" ", "")
-        item = await collection.find_one({"name": re.compile(i, re.IGNORECASE)})
+        itemName = itemName.strip().replace(" ", "")
+        item = await collection.find_one({"name": re.compile(itemName, re.IGNORECASE)})
```

Comment:
Instead of looping through and making multiple database calls, which is very slow, look up the `$in` operator